ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AdminNodeInfoTest.java

```diff
@@ -127,6 +134,58 @@ public void shouldReturnCorrectResult() {
     assertThat(actual.getResult()).isEqualTo(expected);
   }
 
+  @Test
+  public void shouldReturnCorrectResultWhenIsNatEnvironment() {
+    when(p2pNetwork.isP2pEnabled()).thenReturn(true);
+    when(p2pNetwork.getLocalEnode()).thenReturn(Optional.of(defaultPeer.getEnodeURL()));
+
+    when(natService.isNatEnvironment()).thenReturn(true);
+    when(natService.queryExternalIPAddress()).thenReturn(Optional.of("3.4.5.6"));
+    when(natService.getPortMapping(NatServiceType.DISCOVERY, NetworkProtocol.UDP))
+        .thenReturn(
+            Optional.of(
+                new NatPortMapping(
+                    NatServiceType.DISCOVERY, NetworkProtocol.UDP, "", "", 8080, 8080)));
```

Comment:
I just noticed that the only time we return `Optional.empty` is when there was an exception, otherwise, any NAT scheme will have a port mapping (since that's the point of NAT). What do you think about making `getPortMapping` return a `NatPortMapping` and not an `Optional<NatPortMapping>`?

Reply:
You are completely right. I can remove the catch and modify the check of the nat environment so that it is no longer optional. it will mean that each call of this method need to catch an illegal state exception. Personally I prefer to return an `Optional.empty` if this method is called outside the nat environment or if an exception occurs.  if you think it's better to throw an exception I propose an implementation below

Current implementation
```
public Optional<NatPortMapping> getPortMapping(
      final NatServiceType serviceType, final NetworkProtocol networkProtocol) {
    if (isNatEnvironment()) {
      try {
        final NatManager natManager = getNatManager().orElseThrow();
        return Optional.of(natManager.getPortMapping(serviceType, networkProtocol));
      } catch (Exception e) {
        LOG.warn("Caught exception while trying to query port mapping (ignoring): {}", e);
      }
    }
    return Optional.empty();
  }
```

New implementation ?
```
public NatPortMapping getPortMapping(
      final NatServiceType serviceType, final NetworkProtocol networkProtocol)
      throws IllegalStateException {
    checkState(isNatEnvironment(), "Cannot call getPortMapping() outside a NAT environment");
    return getNatManager()
        .orElseThrow(
            () ->
                new IllegalStateException(
                    "Cannot call getPortMapping() without a valid nat manager"))
        .getPortMapping(serviceType, networkProtocol);
  }
```